src/dfx/src/lib/extension/manifest/extension.rs

```diff
@@ -14,15 +16,33 @@ pub struct ExtensionManifest {
     pub categories: Vec<String>,
     pub keywords: Option<Vec<String>>,
     pub description: Option<String>,
-    pub commands: JsonValue,
+    pub subcommands: JsonValue, // TODO: awaiting https://dfinity.atlassian.net/browse/SDK-599
     pub dependencies: Option<HashMap<String, String>>,
 }
 
+#[derive(Debug, Serialize, Deserialize)]
+pub struct ExtensionSubcommand {
+    pub subcommands: Vec<ExtensionSubcommand>,
+    pub key: String,
+    pub summary: String,
+}
+
 impl Display for ExtensionManifest {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        let Ok(json) = serde_json::to_string_pretty(self) else {
+        let Ok(s) = serde_json::to_string_pretty(self) else {
```

Comment:
how did you smuggle let Ok(x) else error?  This should be `.map_err()?`.

Reply:
I was surprised too when I found that last week